libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -682,16 +684,20 @@ proc updateScores(g: GossipSub) = # avoid async
   trace "updating scores", peers = g.peers.len
 
   let now = Moment.now()
-  var evicting: seq[PubSubPeer]
+  var evicting: seq[PeerID]
 
-  for peer, stats in g.peerStats.mpairs:
+  for peerId, stats in g.peerStats.mpairs:
+    let peer = g.peers.getOrDefault(peerId)
```

Comment:
this means we'll never evict disconnected peers from the stats table

Reply:
why? They expire, check a few lines under.